Okta/OktaApi.swift

```diff
@@ -11,27 +11,53 @@
  */
 
 open class OktaApi: NSObject {
-    class func post(_ url: URL, headers: [String: String]?, postData: String?, callback: @escaping ([String: Any]?, OktaError?) -> Void) {
+    class func post(_ url: URL,
+                    headers: [String: String]?,
+                    postData: String?,
+                    callback: @escaping ([String: Any]?, OktaError?) -> Void) {
+
         // Generic POST API wrapper
-        
         var request = URLRequest(url: url)
         request.httpMethod = "POST"
-        request.allHTTPHeaderFields = headers
-        
+        request.allHTTPHeaderFields = headers != nil ? headers : request.allHTTPHeaderFields
+        request.addValue(
+            "okta-sdk-appauth-ios/\(VERSION) iOS/\(UIDevice.current.systemVersion)",
+            forHTTPHeaderField: "X-Okta-User-Agent-Extended"
+        )
+
         if let postBodyData = postData {
             request.httpBody = postBodyData.data(using: .utf8)
         }
-        
-        let task = URLSession.shared.dataTask(with: request){
-            data, response, error in
-            
-            if error != nil {
-                callback(nil, .apiError(error: "\(String(describing: error?.localizedDescription))"))
-                return
+
+        let task = URLSession.shared.dataTask(with: request){ data, response, error in
+            guard let data = data, error == nil else {
+                return callback(nil, .apiError(error: "\(String(describing: error?.localizedDescription))"))
+            }
+            let responseJson = try? JSONSerialization.jsonObject(with: data, options: .mutableContainers) as! [String: Any]
+            return callback(responseJson, nil)
+        }
+        task.resume()
+    }
+
+    class func get(_ url: URL,
+                   headers: [String: String]?,
+                   callback: @escaping ([String: Any]?, OktaError?) -> Void) {
+
+        // Generic GET API wrapper
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.allHTTPHeaderFields = headers != nil ? headers : request.allHTTPHeaderFields
+        request.addValue(
+            "okta-sdk-appauth-ios/\(VERSION) iOS/\(UIDevice.current.systemVersion)",
```

Comment:
Same as above.

Reply:
Could this also be extracted to a method so we are setting this header in a single place?

Reply:
Yep - the request logic is going to be modified when we introduce a breaking change to this library. That's coming in a follow up PR.